prime-router/src/main/kotlin/fhirengine/translation/hl7/utils/ConstantResolver.kt

```diff
@@ -171,22 +176,15 @@ object CustomFHIRFunctions {
         }
     }
 
-    /**
-     * Regex representing the FHIR phone format generated by the linux4health library
-     */
-    private val phoneRegex = """([+](\d{1,3}|1-\d{3}) (\d{3})|\((\d{3})\)) (\d{3}) (\d{4,6})""".toRegex()
-
     /**
      * Gets the phone number country code from the full FHIR phone number stored in
      * the [focus] element.
      * @return a mutable list containing the country code
      */
     fun getPhoneNumberCountryCode(focus: MutableList<Base>): MutableList<Base> {
-        val matchResult = phoneRegex.find(focus[0].primitiveValue())
-        val countryCode = matchResult?.groups?.get(2)?.value?.toInt()
-        return if (countryCode != null) {
-            mutableListOf(IntegerType(countryCode))
-        } else mutableListOf()
+        val primVal = focus[0].primitiveValue()
```

Comment:
Add a check to check for: 1) that is is only one resource in his list, 2) that the resource is a primitive.  If the list is empty I think you will get an exception here.